epuck_ros2_camera/src/camera.cpp

```diff
@@ -87,63 +116,122 @@ class CameraPublisher : public rclcpp::Node {
     return result;
   }
 
-  void timer_callback() {
+  void timerCallback() {
+    auto stamp = now();
+
+    // Initialize V4L2 if needed
     if (mPublisherCompressed->get_subscription_count() > 0 || mPublisherRaw->get_subscription_count()) {
-      if (!mV4l2Initialized) {
-        pipuck_v4l2_init();
-        mV4l2Initialized = true;
-      }
-      pipuck_v4l2_capture(&mCapturedImage);
-    } else if (mV4l2Initialized) {
-      pipuck_v4l2_deinit();
-      mV4l2Initialized = false;
+      // Capture an image
+      initV4l2();
+      pipuck_v4l2_capture(&(mPipuckMmalRgb.input));
+
+      // Publish camera info
+      mCameraInfoMsg.header.stamp = stamp;
+      mPublisherCameraInfo->publish(mCameraInfoMsg);
+    } else {
+      deinitV4l2();
+      return;
```

Comment:
Usually we tried to put the condition with the return at first so that we even don't need to encapsulate the other part in an else.